packages/stable/src/api/geospatial/types.ts

```diff
@@ -0,0 +1,245 @@
+import { InternalId, Limit, ExternalId } from '@cognite/sdk-core';
+
+import { Geometry, GeoJSON } from 'geojson';
+
+export { GeoJSON };
+
+export type GeometryType = 'WKT' | 'GEOJSON';
+
+type GeometryPropertyType =
+  | 'GEOMETRY'
+  | 'POINT'
+  | 'LINESTRING'
+  | 'POLYGON'
+  | 'MULTIPOINT'
+  | 'MULTILINESTRING'
+  | 'MULTIPOLYGON'
+  | 'GEOMETRYCOLLECTION'
+  | 'GEOMETRYZ'
+  | 'POINTZ'
+  | 'LINESTRINGZ'
+  | 'POLYGONZ'
+  | 'MULTIPOINTZ'
+  | 'MULTILINESTRINGZ'
+  | 'MULTIPOLYGONZ'
+  | 'GEOMETRYM'
+  | 'POINTM'
+  | 'LINESTRINGM'
+  | 'POLYGONM'
+  | 'MULTIPOINTM'
+  | 'MULTILINESTRINGM'
+  | 'MULTIPOLYGONM'
+  | 'GEOMETRYCOLLECTIONM'
+  | 'GEOMETRYZM'
+  | 'POINTZM'
+  | 'LINESTRINGZM'
+  | 'POLYGONZM'
+  | 'MULTIPOINTZM'
+  | 'MULTILINESTRINGZM'
+  | 'MULTIPOLYGONZM'
+  | 'GEOMETRYCOLLECTIONZM';
+
+export type GeospatialPropertyType =
+  | 'STRING'
+  | 'LONG'
+  | 'DOUBLE'
+  | 'BOOLEAN'
+  | 'TIMESTAMP'
+  | GeometryPropertyType;
+
+export type GeospatialFeatureTypeProperty =
+  | {
+      type: Exclude<GeospatialPropertyType, GeometryPropertyType | 'STRING'>;
+      description?: string;
+      optional?: boolean;
+    }
+  | {
+      type: GeometryPropertyType;
+      srid?: number;
+      description?: string;
+      optional?: boolean;
+    }
+  | {
+      type: 'STRING';
```

Comment:
why STRING is separated?

Reply:
As we've `size` only for STRING type and srid only with Geometrical types, hence, these clear breakdowns help while sending filters with correct fields being sent with correct type.

Reply:
clear now 👍